shell/weights/update_training_history.py

```diff
@@ -1,54 +1,92 @@
 )
-flags.DEFINE_string("weights_version", None, "The version of these weights (e.g. 'v0')")
 
 FLAGS = flags.FLAGS
 FLAGS(sys.argv)
 
-# Find the single validation logs in the supplied logs dir
-files = os.listdir(FLAGS.logs_path)
-# Tensorboard logs are named based on timestamp, so the largest name alphabetically is the most recent.
-most_recent_file = max(files)
+weights_version = FLAGS.weights_version or input(
+    "Input the weights version for your script\n"
+)
 
-filename = FLAGS.logs_path + "/" + most_recent_file
+training_script_path = FLAGS.training_script_path or input(
+    "Input the path to your training script\n"
+)
+training_script_json_path = training_script_path.replace(".py", ".json")
+full_training_script_path = os.path.abspath(training_script_path)
 
-events = [e for e in summary_iterator(filename)]
+# Build an experiment name structured as dataset/task-version (this experiment name will then match the name of the weights in GCS)
+training_script_rooted_at_training = full_training_script_path[
+    full_training_script_path.index("keras-cv/examples/training/") + 27 :
+]
+training_script_dirs = training_script_rooted_at_training.split("/")
+tensorboard_experiment_name = (
+    f"{training_script_dirs[2]}/{training_script_dirs[0]}-{weights_version}"
+)
 
-max_validation_accuracy = 0
-for event in events[1:]:
-    assert len(event.summary.value) == 1
-    if not event.summary.value[0].tag == "evaluation_accuracy_vs_iterations":
-        continue
-    validation_accuracy = struct.unpack(
-        "f", event.summary.value[0].tensor.tensor_content
-    )[0]
-    max_validation_accuracy = max(max_validation_accuracy, validation_accuracy)
+tensorboard_logs_path = FLAGS.tensorboard_logs_path or input(
+    "Input the path to the TensorBoard logs\n"
+)
+tensorboard_experiment_id = (
+    os.popen(
+        f"tensorboard dev upload --logdir {tensorboard_logs_path} --name {tensorboard_experiment_name} --one_shot --verbose 0"
+    )
+    .read()
+    .split("/")[-2]
+)
+
+tensorboard_experiment = tb.data.experimental.ExperimentFromDev(
+    tensorboard_experiment_id
+)
+
+tensorboard_results = tensorboard_experiment.get_scalars()
 
+training_epochs = max(tensorboard_results[tensorboard_results.run == "train"].step)
+max_validation_accuracy = max(
+    tensorboard_results[
+        (tensorboard_results.run == "validation")
+        & (tensorboard_results.tag == "epoch_accuracy")
+    ].value
+)
 max_validation_accuracy = f"{max_validation_accuracy:.4f}"
 
-print(f"Max validation accuracy: {max_validation_accuracy}")
+new_results = {
+    "validation_accuracy": max_validation_accuracy,
+    "epochs_trained": training_epochs,
+    "tensorboard_logs": f"https://tensorboard.dev/experiment/{tensorboard_experiment_id}/",
+}
 
-new_results = {FLAGS.weights_version: {"validation_accuracy": max_validation_accuracy}}
+pr_number = FLAGS.pr_number or input(
+    "Input the PR number that merged your script into KerasCV\n"
```

Comment:
should this be a link instead of a number?

Reply:
Good call -- I've kept the flag as pr_number but the user input version is a full link.

Reply:
seems reasonable